packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapJSONConversions.m

```diff
@@ -141,4 +141,34 @@ + (nullable GMSCameraUpdate *)cameraUpdateFromChannelValue:(NSArray *)channelVal
   }
   return nil;
 }
+
++ (NSArray<GMSStrokeStyle *> *)strokeStylesFromPatterns:(NSArray *)patterns
+                                            strokeColor:(UIColor *)strokeColor {
+  NSMutableArray *strokeStyles = [[NSMutableArray alloc] init];
+  for (unsigned int i = 0; i < [patterns count]; i++) {
+    NSString *patternType = patterns[i][0];
+    GMSStrokeStyle *strokeStyle;
+
+    if ([patternType isEqualToString:@"gap"]) {
+      strokeStyle = [GMSStrokeStyle solidColor:[UIColor clearColor]];
+    } else {
+      strokeStyle = [GMSStrokeStyle solidColor:strokeColor];
+    }
+
+    [strokeStyles addObject:strokeStyle];
```

Comment:
Consider collapsing this all to:
```objc
UIColor* color = [patternType isEqualToString:@"gap"] ? [UIColor clearColor] : strokeColor;
[strokeStyles addObject:[GMSStrokeStyle solidColor:color];
```

Having mostly identical code in both branches of a conditional is generally a code smell, since it makes it easy for things to accidentally diverge.

Reply:
Great call!